src/sagemaker/jumpstart/types.py

```diff
@@ -1007,81 +1009,74 @@ def to_json(self) -> Dict[str, Any]:
 class JumpStartPresetComponent(JumpStartMetadataBaseFields):
     """Data class of JumpStart config component."""
 
-    slots = ["component_name", "component_override_fields"]
+    slots = ["component_name"]
 
     # List of fields that is not allowed to override to JumpStartMetadataBaseFields
-    # TODO: finalize the fields that can be overrided
-    OVERRIDING_DENY_LIST = []
+    OVERRIDING_DENY_LIST = [
+        "model_id",
+        "url",
+        "version",
+        "min_sdk_version",
+        "deprecated",
+        "deprecated_message",
+        "deprecate_warn_message",
+        "resource_name_base",
+        "gated_bucket",
+        "training_supported",
+        "incremental_training_supported",
+    ]
 
     __slots__ = slots + JumpStartMetadataBaseFields.__slots__
 
-    def __init__(self, component: Optional[Dict[str, Any]], spec: Optional[Dict[str, Any]]):
+    def __init__(
+        self,
+        component_name: str,
+        component: Optional[Dict[str, Any]],
+    ):
         """Initializes a JumpStartInferenceComponent object from its json representation.
 
         Args:
             component (Dict[str, Any]):
-                Dictionary representation of the config component that can override to the metadata base fields.
-            spec (Dict[str, Any]):
-                Dictionary representation of the original metadata base fields.
+                Dictionary representation of the config component.
         """
-        super().from_json(spec)
-        self.component_name: str = component["component_name"]
-        self.component_overrides: JumpStartMetadataBaseFields = JumpStartMetadataBaseFields(
-            component
-        )
+        self.component_name = component_name
+        self.from_json(component)
 
-    def to_json(self) -> Dict[str, Any]:
-        """Returns json representation of JumpStartInferenceComponent object."""
-        json_obj = {att: getattr(self, att) for att in self.__slots__ if hasattr(self, att)}
-        return json_obj
+    def from_json(self, json_obj: Dict[str, Any]) -> None:
```

Comment:
can we not define this in a parent class?

Reply:
Sorry I'm not following, there's no other classes inheriting this class. I did removed this in other parent classes.